Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -6072,6 +6073,26 @@ RefPtr<CSSStyleValue> Parser::parse_quotes_value(TokenStream<ComponentValue>& to
     return StyleValueList::create(move(string_values), StyleValueList::Separator::Space);
 }
 
+RefPtr<CSSStyleValue> Parser::parse_tab_size(TokenStream<ComponentValue>& tokens)
+{
+    auto transaction = tokens.begin_transaction();
+    auto maybe_value = parse_css_value_for_property(PropertyID::TabSize, tokens);
+    if (!maybe_value)
+        return nullptr;
+
+    if (maybe_value->is_length()) {
+        transaction.commit();
+        return maybe_value.release_nonnull();
+    }
+
+    if (maybe_value->is_number()) {
+        transaction.commit();
+        return maybe_value.release_nonnull();
+    }
+
+    return nullptr;
+}
+
```

Comment:
This code looks correct, but it's also unnecessary. :sweat_smile:

`tab-size`'s grammar is `<number [0,∞]> | <length [0,∞]>`, which you've entered in `Properties.json`. Because it's straightforward, the parser can parse it without any specific code here - see the end of `Parser::parse_css_value()`. That calls `parse_css_value_for_property()`, which will reject anything that doesn't match this grammar.

Methods for parsing a specific property are only needed when that property has complicated parsing rules that the automatic code isn't smart enough to handle.

Reply:
Well glad I got it correct even though it's not needed. All a learning experience (when the time comes for another complex parsing rule!)

I'll nuke this and the rest of the areas you've specified down below :)